src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -230,7 +230,7 @@ mp::QemuVirtualMachine::QemuVirtualMachine(const ProcessFactory* process_factory
                      [this](int exitCode, QProcess::ExitStatus exitStatus) {
                          mpl::log(mpl::Level::info, vm_name,
                                   fmt::format("process finished with exit code {}", exitCode));
-                         if (update_shutdown_status)
+                         if (update_shutdown_status || state == State::starting)
```

Comment:
`update_shutdown_status` is not new to this PR, so I think we should defer that discussion for later...unless I misunderstand what you mean here.

Reply:
I guess my point is that changes here complexify an already intricate flag: `update_shutdown_status` used to mean approximately whether we want to record the shutdown or suspended state (IIUC). Now it doesn't anymore, because it may be `false` but we still want to do that. 

After your comment below, I understand the change is about not recording the shutdown state when there is an error while starting. Wouldn't it be better then to change the `if` in the error case to `if (update_shutdown_status && state != State::starting)` and leave `update_shutdown_state` alone here and below?

Additionally, in the error handler this flag is checked for a different purpose that what its name says. Couldn't the `if` there be based solely on state?

Reply:
Ok, I'll think about this some more and take your point into consideration.

Reply:
I think @ricab has a point here, our state machine is getting hard to follow. It does need rethinking, but definitely not here.

I follow the intent of this line change, so I'll ok it.